redis_session.c

```diff
@@ -183,6 +186,130 @@ redis_pool_get_sock(redis_pool *pool, const char *key TSRMLS_DC) {
     return NULL;
 }
 
+void lock_acquire(RedisSock *redis_sock, redis_session_lock_status *lock_status)
+{
+    int locking_enabled = INI_INT("redis.session.locking_enabled");
+
+    if (!lock_status->is_locked && locking_enabled > 0) {
+        char *cmd, *response;
+        int response_len, cmd_len, lock_wait_time, max_lock_retries, i_lock_retry, lock_expire;
+        smart_string lock_key = {0};
+
+        calculate_secret_hash(lock_status);
+
+        lock_wait_time = INI_INT("redis.session.lock_wait_time");
+        if (lock_wait_time == 0) {
+          lock_wait_time = 2000;
+        }
+
+        max_lock_retries = INI_INT("redis.session.lock_retries");
+        if (max_lock_retries == 0) {
+          max_lock_retries = 10;
+        }
+
+        lock_expire = INI_INT("redis.session.lock_expire");
+        if (lock_expire == 0) {
+          lock_expire = INI_INT("max_execution_time");
+        }
+
+        // Building the redis lock key
+        smart_string_appendl(&lock_key, lock_status->session_key, strlen(lock_status->session_key));
+        smart_string_appendl(&lock_key, "_LOCK", strlen("_LOCK"));
+        smart_string_0(&lock_key);
+        strncpy(lock_status->lock_key, lock_key.c, 256);
+
+        if (lock_expire > 0) {
+            cmd_len = REDIS_SPPRINTF(&cmd, "SET", "ssssd", lock_status->lock_key, strlen(lock_status->lock_key), lock_status->lock_secret_hash, strlen(lock_status->lock_secret_hash), "NX", 2, "PX", 2, lock_expire * 1000);
```

Comment:
Is the multiplication to 1K necessary here? May be `redis.session.lock_expire` should be in milliseconds?